### DISCUSSION TITLE
Audio NOOB

### DISCUSSION THREAD

**Discussion by Rochambeaux** [mcguirepr89/BirdNET-Pi#458] [Q&A]
Is it possible to "calibrate" the microphone setup on the BirdNET Pi?

Can anyone recommend good beginner resources to understand what I'm looking at in the Spectrogram windows and what adjustments might be needed?

![image](https://user-images.githubusercontent.com/86641453/179628745-fee46de7-d04a-4aaf-9260-4af44481938b.png)

**Comment by ehpersonal38** [comment]
What do you mean by calibrate? You're able to adjust the gain (volume) of the microphone. You can find a tutorial here: https://github.com/mcguirepr89/BirdNET-Pi/wiki/Adjusting-your-sound-card

The vertical axis of the spectrogram displays the frequency in kHZ of sound at that moment in time. The higher the number, the higher the pitch. For each frequency, a color is drawn corresponding to the loudness of the sound. The ligher the color, the louder the volume. This is labeled as dBFS in the bottom right.

Actually I think it's _more_ confusing when only explained. Are you able to access the live Spectrogram? (Note: I _just_ pushed an update that will hopefully allow you to use it on Mac with Chrome, go to Tools -> System Controls -> Update and try it out.) You can play around with different inputs to the microphone and see how they affect it.

**Reply by Rochambeaux** [reply]
> What do you mean by calibrate? You're able to adjust the gain (volume) of the microphone. You can find a tutorial here: https://github.com/mcguirepr89/BirdNET-Pi/wiki/Adjusting-your-sound-card
> 
> The vertical axis of the spectrogram displays the frequency in kHZ of sound at that moment in time. The higher the number, the higher the pitch. For each frequency, a color is drawn corresponding to the loudness of the sound. The ligher the color, the louder the volume. This is labeled as dBFS in the bottom right.
> 
> Actually I think it's _more_ confusing when only explained. Are you able to access the live Spectrogram? (Note: I _just_ pushed an update that will hopefully allow you to use it on Mac with Chrome, go to Tools -> System Controls -> Update and try it out.) You can play around with different inputs to the microphone and see how they affect it.

Thanks.

tried to update and get this:

fatal: cannot change to '/var/lib/caddy/BirdNET-Pi/scripts/BirdNET-Pi': No such file or directory
fatal: cannot change to '/var/lib/caddy/BirdNET-Pi/scripts/BirdNET-Pi': No such file or directory
fatal: cannot change to '/var/lib/caddy/BirdNET-Pi/scripts/BirdNET-Pi': No such file or directory
## PRIVACY_MODE can be set to 'on' or 'off' to configure analysis to be more
## sensitive to human detections. PRIVACY_MODE 'on' will purge any data that
PRIVACY_MODE=off

**Reply by ehpersonal38** [reply]
Ouch, sorry about that! Try:

1. Go to "Tools" > "Web Terminal" OR you any way you can get a console to your Pi to execute commands
2. Type ```git pull```
3. Go to "Tools" > "System Controls" > "Update" > "OK"

**Reply by Rochambeaux** [reply]
> Ouch, sorry about that! Try:
> 
>     1. Go to "Tools" > "Web Terminal" OR you any way you can get a console to your Pi to execute commands
> 
>     2. Type `git pull`
> 
>     3. Go to "Tools" > "System Controls" > "Update" > "OK"

Thanks.
![image](https://user-images.githubusercontent.com/86641453/179640481-474609f4-028e-4621-9598-717cdf972cdc.png)

Update results in same message above.

**Reply by ehpersonal38** [reply]
Can you provide the output of 
`echo $(awk -F: '/1000/ {print $6}' /etc/passwd)`

**Reply by Rochambeaux** [reply]
> Can you provide the output of `echo $(awk -F: '/1000/ {print $6}' /etc/passwd)`

![image](https://user-images.githubusercontent.com/86641453/179869920-921fb80f-0b5a-4fcc-bb72-faf29adfb912.png)

**Reply by ehpersonal38** [reply]
I'm not really sure what's wrong, never seen this error before. Sorry this issue spiraled into another. You can try running `bash ~/BirdNET-Pi/scripts/update_birdnet.sh`

**Reply by Rochambeaux** [reply]
> 

No worries, thanks for your continued help:

![image](https://user-images.githubusercontent.com/86641453/180013241-e2fed9a0-046f-4c46-9c22-3a986acec213.png)

**Reply by ehpersonal38** [reply]
@mcguirepr89 I bet you'd know what's up here?

**Reply by mcguirepr89** [reply]
There is definitely a spanner in the works!!

I would try:
```
git checkout -f
git pull -f
~/BirdNET-Pi/scripts/update_birdnet.sh
```
to be sure that the update script is not the issue itself.

Let us know if that results in a different outcome than issuing the update script alone.

**Reply by Rochambeaux** [reply]
> There is definitely a spanner in the works!!
> 
> I would try:
> 
> ```
> git checkout -f
> git pull -f
> ~/BirdNET-Pi/scripts/update_birdnet.sh
> ```
> 
> to be sure that the update script is not the issue itself.
> 
> Let us know if that results in a different outcome than issuing the update script alone.

![image](https://user-images.githubusercontent.com/86641453/180030220-507a3f4c-343b-45ce-9dd8-1c9a39387434.png)

**Reply by ehpersonal38** [reply]
Ah, I guess that would do it. It looks like you're using the CoreElectronics repo (I noticed it says `origin/stable`). Where do we go from here? @mcguirepr89

**Reply by mcguirepr89** [reply]
~:thinking: that is strange indeed -- something is quite wrong:~

~Could you please run this:~
```
sed '3 i set -x' scripts/update_birdnet.sh | bash 2>&1 | sed '/PWD/d;/KEY/d;/LATITUDE/d;/LONGITUDE/d;/URL/d;/STREAM/d;/BIRDWEATHER/d'
```

~I'm hopeful we'll see the error somewhere within that debugging (I've attempted to scrub info that hasn't already been shared, like your LAT, LON, passwords, API keys, URLs, etc)~

**Reply by mcguirepr89** [reply]
> Ah, I guess that would do it. It looks like you're using the CoreElectronics repo (I noticed it says origin/stable). Where do we go from here? @mcguirepr89

FANTASTIC CATCH!!! THANK YOU!!!

From here, what do we do? We can try:

```
git remote set-remote origin https://github.com/mcguirepr89/BirdNET-Pi.git
git checkout -f main
git pull
update_birdnet.sh
```

I _think_ that will work

**Reply by Rochambeaux** [reply]
> > Ah, I guess that would do it. It looks like you're using the CoreElectronics repo (I noticed it says origin/stable). Where do we go from here? @mcguirepr89
> 
> FANTASTIC CATCH!!! THANK YOU!!!
> 
> From here, what do we do? We can try:
> 
> ```
> git remote set-remote origin https://github.com/mcguirepr89/BirdNET-Pi.git
> git checkout -f main
> git pull
> update_birdnet.sh
> ```
> 
> I _think_ that will work

![image](https://user-images.githubusercontent.com/86641453/180043395-45652e67-a5fd-4d5c-beb1-0414cb8d1a77.png)

**Reply by mcguirepr89** [reply]
sorry, I coded from the hip on that one --

```
git remote set-url origin https://github.com/mcguirepr89/BirdNET-Pi.git
git checkout -f main
git pull
update_birdnet.sh
```

**Reply by Rochambeaux** [reply]
> update_birdnet.sh

Success??
![image](https://user-images.githubusercontent.com/86641453/180066930-cf233cd0-8036-42b7-9a41-1ea768fbe696.png)

![image](https://user-images.githubusercontent.com/86641453/180067399-1356ac75-f4c6-4d26-af05-9324a30d796c.png)

![image](https://user-images.githubusercontent.com/86641453/180067431-73616596-84ca-44b9-aa78-5531645ac113.png)

**Reply by mcguirepr89** [reply]
That certainly looks a lot better.

How are your logs looking in "View Logs"?

**Reply by Rochambeaux** [reply]
> That certainly looks a lot better.
> 
> How are your logs looking in "View Logs"?

![image](https://user-images.githubusercontent.com/86641453/180071915-64e32f95-f60f-4835-af28-30a1eb7aece1.png)

**Reply by mcguirepr89** [reply]
Try restarting services (your system underwent a TONNNNN of updates from the CoreElectronics version)

"Tools" > "Services" > "Restart Core Services" and let me know if that helps revive the system

**Reply by ehpersonal38** [reply]
![image](https://user-images.githubusercontent.com/103586016/180072583-970c34bd-0146-423a-b664-6f2a6ba1e84f.png)
That "Terminated" seems questionable, hopefully it's nothing though 😅

**Reply by mcguirepr89** [reply]
Ahhh the Terminated is from the update restarting the web terminal
